featuretools/entityset/serialization.py

```diff
@@ -1,175 +1,177 @@
-# -*- coding: utf-8 -*-
-
 import json
 import os
 import shutil
-import sys
-import uuid
-from tempfile import mkdtemp
 
-import numpy as np
 import pandas as pd
-from pandas.io.pickle import read_pickle as pd_read_pickle
-from pandas.io.pickle import to_pickle as pd_to_pickle
 
+from .. import variable_types
+
+TYPES = ['csv', 'pickle', 'parquet']
+SCHEMA = {
+    'entities':
+    ['id', 'index', 'time_index', 'variables', 'properties', 'loading_info'],
+    'relationships': {
+        'parent': ['parent_entity', 'parent_variable'],
+        'child': ['child_entity', 'child_variable']
+    }
+}
+VTYPES = {
+    getattr(variable_types, v)._dtype_repr: getattr(variable_types, v)
+    for v in dir(variable_types)
+    if hasattr(getattr(variable_types, v), '_dtype_repr')
+}
+
+
+def to_entity_descr(e):
+    '''Serialize entity to data description.
+
+    Args:
+        e (Entity) : Instance of :class:`.Entity`.
+
+    Returns:
+        item (tuple(str, dict)) : Tuple containing id (str) and description (dict) of :class:`.Entity`.
+    '''
+    descr = {}
+    for key in SCHEMA.get('entities'):
+        if key in ['id', 'index', 'time_index']:
+            descr[key] = getattr(e, key)
+        elif key == 'properties':
+            sti, lti = e.secondary_time_index, e.last_time_index is not None
+            descr[key] = dict(secondary_time_index=sti, last_time_index=lti)
+        elif key == 'variables':
+            descr[key] = [v.create_data_description() for v in e.variables]
+        elif key == 'loading_info':
+            dtypes = dict(dtypes=e.df.dtypes.astype(str).to_dict())
+            descr[key] = dict(properties=dtypes, params={})
+        else:
+            raise ValueError('"{}" is not supported'.format(key))
+    return e.id, descr
+
+
+def to_relation_descr(r):
```

Comment:
no need to abbreviate. just use full name for clarity here `to_relationship_description`

Reply:
Done